ipc/provider/src/checkpoint.rs

```diff
@@ -162,30 +178,71 @@ impl<T: BottomUpCheckpointRelayer + Send + Sync + 'static> BottomUpCheckpointMan
                     .await?;
                 log::debug!("bottom up bundle: {bundle:?}");
 
-                let epoch = self
-                    .parent_handler
-                    .submit_checkpoint(
-                        submitter,
-                        bundle.checkpoint,
-                        bundle.signatures,
-                        bundle.signatories,
-                    )
+                // We support parallel checkpoint submission using FIFO order with a limited parallelism (controlled by
+                // the size of submission_semaphore).
+                // We need to acquire a permit (from a limited permit pool) before submitting a checkpoint.
+                // We may wait here until a permit is available.
+                let parent_handler_clone = Arc::clone(&self.parent_handler);
+                let submission_permit = self
+                    .submission_semaphore
+                    .clone()
+                    .acquire_owned()
                     .await
-                    .map_err(|e| {
-                        anyhow!(
-                            "cannot submit bottom up checkpoint at height {} due to: {e:}",
-                            event.height
-                        )
-                    })?;
-
-                log::info!(
-                    "submitted bottom up checkpoint({}) in parent at height {}",
-                    event.height,
-                    epoch
+                    .unwrap();
+                all_submit_tasks.push(tokio::task::spawn(async move {
+                    match Self::submit_checkpoint(parent_handler_clone, submitter, bundle, event)
+                        .await
+                    {
+                        Ok(()) => {
+                            log::debug!("Successfully submitted checkpoint");
```

Comment:
Maybe log which height has been submitted.

Reply:
I see this is completely redundant with the `info` log [below](https://github.com/consensus-shipyard/ipc/pull/840/files#diff-43df7702e44369062dcd87a6d917f2d9e7a026ef31e8bacedd1248088b064b65R241). 

I would suggest one of two options:
* only do logging here, remove the one below, or
* remove the success logging from here and keep the one below, logging only the errors here: `if let Err(e) = Self::submit_checkpoint(...) { log::error!(error=e, ... }`